extensions/postgres/transactional.go

```diff
@@ -15,17 +15,37 @@ package postgres
 
 import (
 	"context"
+
 	"github.com/xdblab/xdb/common/uuid"
 	"github.com/xdblab/xdb/extensions"
 )
 
-const insertCurrentProcessExecutionQuery = `INSERT INTO xdb_sys_current_process_executions
+const insertLatestProcessExecutionQuery = `INSERT INTO xdb_sys_latest_process_executions
 	(namespace, process_id, process_execution_id) VALUES
 	($1, $2, $3)`
 
-func (d dbTx) InsertCurrentProcessExecution(ctx context.Context, row extensions.CurrentProcessExecutionRow) error {
+func (d dbTx) InsertLatestProcessExecution(ctx context.Context, row extensions.LatestProcessExecutionRow) error {
 	row.ProcessExecutionIdString = row.ProcessExecutionId.String()
-	_, err := d.tx.ExecContext(ctx, insertCurrentProcessExecutionQuery, row.Namespace, row.ProcessId, row.ProcessExecutionIdString)
+	_, err := d.tx.ExecContext(ctx, insertLatestProcessExecutionQuery, row.Namespace, row.ProcessId, row.ProcessExecutionIdString)
+	return err
+}
+
+const selectLatestProcessExecutionForUpdateQuery = `SELECT namespace, process_id, process_execution_id 
+FROM xdb_sys_latest_process_executions 
+WHERE namespace=$1 AND process_id=$2 FOR UPDATE`
+
+func (d dbTx) SelectLatestProcessExecutionForUpdate(
+	ctx context.Context, namespace string, processId string,
+) ([]extensions.LatestProcessExecutionRow, error) {
+	var rows []extensions.LatestProcessExecutionRow
+	err := d.tx.SelectContext(ctx, &rows, selectLatestProcessExecutionForUpdateQuery, namespace, processId)
```

Comment:
If it's expecting a single row, then use GetContext to ensure only single row is returned. https://jmoiron.github.io/sqlx/

Reply:
right, here I use selectContext is because there could be 0 or 1 row is returned, getContext will consider 0 row is an error, thus abort the transaction

Reply:
I see. It’s probably better to return (Row, bool, error) and use the bool to indicate empty.